nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -896,6 +893,22 @@ in
           '';
       };
 
+      typesHashMaxSize = mkOption {
+        type = types.ints.positive;
+        default = if cfg.defaultMimeTypes == "${pkgs.mailcap}/etc/nginx/mime.types" then 2688 else 1024;
+        defaultText = literalExpression ''if cfg.defaultMimeTypes == "''${pkgs.mailcap}/etc/nginx/mime.types" then 2688 else 1024'';
+        description = ''
+          Sets the maximum size of the types hash tables.
+
+          When recommendedOptimisation is disabled nginx fails to start because the
+          mailmap mime.types database contains 1026 entries and the default is only
+          1024. Setting to a higher number to remove the need to overwrite it because
+          nginx does not allow duplicated settings.
+
+          It is recommended that the minimum size be used.
```

Comment:
```suggestion
          Sets the maximum size of the types hash tables (`types_hash_max_size`).
          It is recommended that the minimum size possible size is used.

          If {option}`recommendedOptimisation` is disabled, nginx would otherwise
          fail to start since the mailmap `mime.types` database has more entries
          than the nginx default value 1024.
```

Reply:
Thanks, fixed.